frontend/packages/dev-console/src/components/project-access/project-access-form-utils.ts

```diff
@@ -0,0 +1,166 @@
+import * as yup from 'yup';
+import * as _ from 'lodash';
+import { k8sCreate, k8sKill, k8sPatch, K8sResourceKind } from '@console/internal/module/k8s';
+import { RoleBindingModel } from '@console/internal/models';
+import { generateSecret } from '../import/import-submit-utils';
+
+export enum Verb {
+  Create = 'Create',
+  Remove = 'Remove',
+  Patch = 'Patch',
+}
+
+export enum Roles {
+  view = 'view',
+  admin = 'admin',
+}
+
+export interface UserRole {
+  metadata: {
+    name: string;
+  };
+  roleRef: {
+    name: string;
+  };
+  subjects: [
+    {
+      name: string;
+    }
+  ];
+}
+
+export interface UserRoleBinding {
+  roleBindingName?: string;
+  user: string;
+  role: string;
+}
+
+export const generateRoleBindingName = (username: string, role: string) => {
+  return `${username}-${role}-${generateSecret()}`;
+};
+
+export const sendK8sRequest = (verb: string, roleBinding): Promise<K8sResourceKind> => {
+  switch (verb) {
+    case Verb.Create:
+      return k8sCreate(RoleBindingModel, roleBinding);
+    case Verb.Remove:
+      return k8sKill(RoleBindingModel, roleBinding);
+    case Verb.Patch:
+      return k8sPatch(RoleBindingModel, { metadata: roleBinding.metadata }, [
+        { op: 'replace', path: `/subjects/0`, value: roleBinding.subjects[0] },
+      ]);
+    default:
+      return null;
+  }
+};
+
+export const sendRoleBindingRequest = (
+  verb: string,
+  formName: string,
+  roles: UserRoleBinding[],
+  finalRoles: UserRoleBinding[],
+  roleBinding,
+  actions,
+) => {
+  const finalArray = [];
+  _.forEach(roles, (user) => {
+    const roleBindingName =
+      verb === Verb.Create ? generateRoleBindingName(user.user, user.role) : user.roleBindingName;
+    roleBinding.subjects[0].name = user.user;
+    roleBinding.roleRef.name = user.role;
+    roleBinding.metadata.name = roleBindingName;
+    finalArray.push(sendK8sRequest(verb, roleBinding));
+  });
+  Promise.all(finalArray)
```

Comment:
I think this part should be in `handleSubmit` or a function in the component since it uses actions from the formik context. You can just create the request here and send back the promise, then handle submit can execute that promise using `Promise.all`.